springwolf-core/src/test/java/io/github/stavshamir/springwolf/configuration/DefaultAsyncApiDocketServiceIntegrationTest.java

```diff
@@ -32,6 +32,7 @@ public class DefaultAsyncApiDocketServiceIntegrationTest {
                 "springwolf.enabled=true",
                 "springwolf.docket.info.title=Info title was loaded from spring properties",
                 "springwolf.docket.info.version=1.0.0",
+                "springwolf.docket.info.extension-fields.x-api-name=api-name",
```

Comment:
Test of info extension

Reply:
👍 I would like to see that these fields are also serialized into the json document.
We use the example projects to demonstrate, test and verify features. Can you add a field to an example (i. e. amqp) and update the Asyncapi.json?

Reply:
I think there was a misunderstanding and you did much more than we would have expected. We do not need to test the manually defined docket vs the docket build from spring properties again in an example project. This should be/is already done in core. I think what Timon actually meant: We can just configure extension fields in the application properties (like you already did) and just add those to the original asyncapi.json file :). Sorry for the confusion but I think it makes your change much simpler...

Reply:
I am not quite sure what you mean :slightly_smiling_face: . Do you mean to replace the custom springwolf `Info` class in `SpringWolfConfigProperties` with the `com.asyncapi...Info` class, so that properties from application.yml are mapped directly to the 'info' part of the asyncapi?  

If this is the case: I wondered already why `SpringWolfConfigProperties` uses  `com.asyncapi...Contact` and `com.asyncapi...License` while `Info` is implemented as custom springwolf class.  I assumed, the different 'Nullable' contracts of `title` and `version` were the reason for this design decision. (?)

If I replace springwolf `Info` with `com.asyncapi...Info` there will be a slightly different contract, title and version have to be provided than in application.yml.

Please give me a quick signal if this is what you mean and whether I can simply the SpringWolfConfigPropertis Info part...

Reply:
No that's not what I meant😅. I will try to write it down more clearly tomorrow...

Reply:
Hi @tvahrst, so my original comment was about the number of test cases. We would only have expected the change you did in AsyncApiConfiguration, so that users can see how the spec would look like with the extension fields. I had another look at the code and especially the setup of the example, and I think I can now understand why you added the second test case where the extensions would be set in application.properties. As it is just an example project for demonstration purposes, we do not want to do extensive testing of core functionality there to keep maintenance efforts low.

I can see the point of adding the example where the extensions are set by spring properties instead of static configuration in the code, because thats the feature you wanted to add. I would suggest to keep the PR as it is for now, I will discuss with Timon how we want to deal with this cases and we will then either merge the PR as is or remove one of test cases. Is that okay for you?

Reply:
Of course, thats ok. 👍